### DISCUSSION TITLE
How to write unit tests for ViewModels that have routers

### DISCUSSION THREAD

**Discussion by JamesSedlacek** [SwiftfulThinking/SwiftfulRouting#15] [Q&A]
Can somebody create an example of how to write a unit test that has a router in it?

**Comment by SwiftfulThinking** [comment]
@JamesSedlacek What specifically are you trying to test?

**Reply by JamesSedlacek** [reply]
I just want to write unit tests for ViewModels, but I'm not sure how to handle the router property:
`private let router: AnyRouter`

**Reply by SwiftfulThinking** [reply]
Ok, I need to write tests for the routing itself, but you should be able to test your ViewModel independently. `AnyRouter` is a type-erased `Router`. So you can make a `MockRouter` that conforms to `Router` and then use it for testing purposes.

It would be something like:

```swift
struct MockRouter: Router {
    // Router methods
}

// For testing
let router = AnyRouter(object: MockRouter())
```

**Reply by JamesSedlacek** [reply]
This makes sense! I'll give it a try, thanks!
